### PR TITLE
Closes #2590 - Deleted users makes it impossible to download CSV cont…

### LINKED ISSUE #2590: Deleted users makes it impossible to download CSV containing users' list
Superuser has the ability to download CSV with users' data. This works fine in all installations except the one where we have a deleted user and where the app throws an error: 

`{"status":"ERROR","message":"Internal server error. More details available at the server logs.","code":500,"incidentId":"97b0b3f3-ca6e-4568-be60-b452e10724ee"}`

Logs: 

```
[2024-11-19T14:03:27.466+0000] [glassfish 4.1] [WARNING] [] [org.eclipse.persistence.session./file:/opt/glassfish4/glassfish/domains/domain1/applications/dataverse/WEB-INF/lib/dataverse-persistence-1.0.0-SNAPSHOT.jar_VDCNet-ejbPU] [tid: _ThreadID=17297 _ThreadName=http-listener-1(53)] [timeMillis: 1732025007466] [levelValue: 900] [[

javax.validation.ConstraintViolationException: Bean Validation constraint(s) violated while executing Automatic Bean Validation on callback event:'preUpdate'. Please refer to embedded ConstraintViolations for details.
        at org.eclipse.persistence.internal.jpa.metadata.listeners.BeanValidationListener.validateOnCallbackEvent(BeanValidationListener.java:108)
        at org.eclipse.persistence.internal.jpa.metadata.listeners.BeanValidationListener.preUpdate(BeanValidationListener.java:87)
        at org.eclipse.persistence.descriptors.DescriptorEventManager.notifyListener(DescriptorEventManager.java:721)
        at org.eclipse.persistence.descriptors.DescriptorEventManager.notifyEJB30Listeners(DescriptorEventManager.java:691)
        at org.eclipse.persistence.descriptors.DescriptorEventManager.executeEvent(DescriptorEventManager.java:229)
        at org.eclipse.persistence.descriptors.changetracking.DeferredChangeDetectionPolicy.calculateChanges(DeferredChangeDetectionPolicy.java:87)
        at org.eclipse.persistence.descriptors.changetracking.DeferredChangeDetectionPolicy.calculateChangesForExistingObject(DeferredChangeDetectionPolicy.java:56)
        at org.eclipse.persistence.internal.sessions.UnitOfWorkImpl.calculateChanges(UnitOfWorkImpl.java:664)
        at org.eclipse.persistence.internal.sessions.UnitOfWorkImpl.commitToDatabaseWithChangeSet(UnitOfWorkImpl.java:1516)
        at org.eclipse.persistence.internal.sessions.UnitOfWorkImpl.issueSQLbeforeCompletion(UnitOfWorkImpl.java:3168)
        at org.eclipse.persistence.internal.sessions.RepeatableWriteUnitOfWork.issueSQLbeforeCompletion(RepeatableWriteUnitOfWork.java:355)
        at org.eclipse.persistence.transaction.AbstractSynchronizationListener.beforeCompletion(AbstractSynchronizationListener.java:158)
        at org.eclipse.persistence.transaction.JTASynchronizationListener.beforeCompletion(JTASynchronizationListener.java:68)
        at com.sun.enterprise.transaction.JavaEETransactionImpl.commit(JavaEETransactionImpl.java:452)
        at com.sun.enterprise.transaction.JavaEETransactionManagerSimplified.commit(JavaEETransactionManagerSimplified.java:854)
        at com.sun.ejb.containers.EJBContainerTransactionManager.completeNewTx(EJBContainerTransactionManager.java:723)
        at com.sun.ejb.containers.EJBContainerTransactionManager.postInvokeTx(EJBContainerTransactionManager.java:507)
        at com.sun.ejb.containers.BaseContainer.postInvokeTx(BaseContainer.java:4566)
        at com.sun.ejb.containers.BaseContainer.postInvoke(BaseContainer.java:2074)
        at com.sun.ejb.containers.BaseContainer.postInvoke(BaseContainer.java:2044)
        at com.sun.ejb.containers.EJBLocalObjectInvocationHandler.invoke(EJBLocalObjectInvocationHandler.java:220)
        at com.sun.ejb.containers.EJBLocalObjectInvocationHandlerDelegate.invoke(EJBLocalObjectInvocationHandlerDelegate.java:88)
        at com.sun.proxy.$Proxy543.findAllAuthenticatedUsers(Unknown Source)
        at edu.harvard.iq.dataverse.authorization.__EJB31_Generated__AuthenticationServiceBean__Intf____Bean__.findAllAuthenticatedUsers(Unknown Source)
        at edu.harvard.iq.dataverse.api.Users.lambda$listUsersCSV$0(Users.java:41)
        at org.glassfish.jersey.message.internal.StreamingOutputProvider.writeTo(StreamingOutputProvider.java:78)
        at org.glassfish.jersey.message.internal.StreamingOutputProvider.writeTo(StreamingOutputProvider.java:60)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.invokeWriteTo(WriterInterceptorExecutor.java:265)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:250)
        at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162)
        at org.glassfish.jersey.server.internal.JsonWithPaddingInterceptor.aroundWriteTo(JsonWithPaddingInterceptor.java:106)
```

The record of the deleted user looks like this:

```
repod_db=> select id, useridentifier, email, firstname, lastname, createdtime, lastlogintime from authenticateduser a where a.id = 9999;

  id  | useridentifier | email | firstname | lastname |     createdtime     |    lastlogintime
------+----------------+-------+-----------+----------+---------------------+---------------------
 9999 | ERASED_001     |       |           |          | 1970-01-01 00:00:00 | 1970-01-01 00:00:00
```

* The above id is fake.

Probably the exporter is (too) eagerly trying to validate things and when it finds something unexpected (empty name?), it throws an error.

As the list visible on the dashboard page looks fine, maybe validating things in the same way for the CSV exporter will solve the problem.

### PR DESCRIPTION
**Pull request by lbownik**
…aining users' list

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardUserInfoService.java [hunk 1]
```diff
@@ -27,8 +27,13 @@ public class DashboardUserInfoService {
     public List<DashboardUserInfo> createDashboardUsers(UserListResult userListResult) {
         List<DashboardUserInfo> userInfoList = new LinkedList<>();
         for (AuthenticatedUser user : userListResult.getUserList()) {
-            userInfoList.add(new DashboardUserInfo(user, getAuthProviderFriendlyName(user),
-                    hasSelectedConfirmedEmail(user), getUserNotificationLanguageDisplayName(user, session.getLocale())));
+            if (!user.isErased()) {
```
Comment (diasf) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardUserInfoService.java]: Not sure this is the best place to filter. What about counters, are they still correct? I would have guessed that `UserServiceBean` or queries in `AuthenticatedUser` would be a better place to filter out erased accounts in order have a more consistent impact.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/userdata/AuthenticatedUserCsvWriter.java [hunk 2]
```diff
@@ -40,7 +40,9 @@ public void write(OutputStream outputStream, List<AuthenticatedUser> authenticat
 
             csvPrinter.printRecord(AuthenticatedUserCSVRecord.getHeaders());
             for(AuthenticatedUser user : authenticatedUsers) {
-                csvPrinter.printRecord(buildRecord(user).getValues());
+                if(! user.isErased()) {
```
Comment (diasf) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/userdata/AuthenticatedUserCsvWriter.java]: Same as above, this seems to be the wrong place to filter out records.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardUserInfoService.java (hunk 1 above)
Comment (lbownik) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dashboard/DashboardUserInfoService.java]: fixed

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/userdata/AuthenticatedUserCsvWriter.java (hunk 2 above)
Comment (lbownik) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/userdata/AuthenticatedUserCsvWriter.java]: fixed

**Review by diasf — APPROVED** [review]
LGTM

**Pull request merged**